DOC: API reference tweaks & clean-up

benbovy opened the pull request:
Mostly cosmetic changes to make the API reference documentation nicer and more consistent across the library.

More details in each commit message.

Also includes changes suggested in #87.

jorisvandenbossche commented on docs/conf.py:
```diff
@@ -18,8 +18,11 @@
+    "pyarrow": ("https://arrow.apache.org/docs/", None),
 }
 
+# TODO: seem to have no effect
+# (see https://github.com/sphinx-doc/sphinx/issues/10235)
```
This comment is now no longer up to date? 
(the commit message in https://github.com/benbovy/spherely/pull/89/commits/c59675e82d612d9bcadd677540dd260a4c7ad9ee seemed to indicate that you changed to manual signatures so `autodoc_typehints = "none"` would actually work?)

benbovy replied:
I think we can just remove `autodoc_typehints = "none"` now that we have switched to manual signatures globally.

The behavior is a bit confusing. IIUC when both `autodoc_docstring_signature=True` (default) *and* the 1st line of the docstrings looks like a signature, then Sphinx pops the 1st line and overrides the signature ignoring `autodoc_typehints` in the process (from https://github.com/sphinx-doc/sphinx/issues/10235#issuecomment-1059896282).

jorisvandenbossche commented on src/geoarrow.cpp:
```diff
@@ -255,20 +255,25 @@ ArrowArrayHolder to_geoarrow(py::array_t<PyObjectGeography> input,
+        py::arg("tessellate_tolerance") = 100.0,
+        py::arg("projection") = Projection::lnglat(),
+        py::arg("geometry_encoding") = py::none(),
+        R"pbdoc(from_geoarrow(geographies, /, *, oriented=False, planar=False, tessellate_tolerance=100.0, projection=Projection.lnglat(), geometry_encoding=None)
```
```suggestion
        R"pbdoc(from_geoarrow(geographies, /, *, oriented=False, planar=False, tessellate_tolerance=100.0, projection=spherely.Projection.lnglat(), geometry_encoding=None)
```
?

jorisvandenbossche commented on src/geoarrow.cpp:
```diff
@@ -315,28 +320,36 @@ void init_geoarrow(py::module& m) {
+        py::arg("planar") = false,
+        py::arg("tessellate_tolerance") = 100.0,
+        py::arg("precision") = 6,
+        R"pbdoc(to_geoarrow(geographies, /, *, output_schema=None, projection=Projection.lnglat(), planar=False, tessellate_tolerance=100.0, precision=6)
```
```suggestion
        R"pbdoc(to_geoarrow(geographies, /, *, output_schema=None, projection=spherely.Projection.lnglat(), planar=False, tessellate_tolerance=100.0, precision=6)
```

jorisvandenbossche commented on src/geography.cpp:
```diff
@@ -318,36 +345,92 @@ void init_geography(py::module &m) {
     pygeography.def(py::pickle([](Geography &geog) { return geog.encode(); },
                                [](py::tuple &encoded) { return Geography::decode(encoded); }));
 
+    py::options options;
+    options.disable_function_signatures();
+
     // Geography properties
 
     m.def("get_type_id",
           py::vectorize(&get_type_id),
           py::arg("geography"),
-          R"pbdoc(
-        Returns the type ID of a geography.
+          R"pbdoc(get_type_id(geography)
 
-        - None (missing) is -1
-        - POINT is 0
-        - LINESTRING is 1
+        Returns the type ID of a geography.
```
I would personally keep this list here (but then expanded to be correct). Of course that gives two places we have to keep this up-to-date (although hopefully we won't change those values often after 0.1.0 ..). 
When using this function, it's nice to directly see what the return values mean without having to first check another docstring (when using the online documentation it is of course only a single click, but when in an interactive python session it is a bit more convoluted)

Or at least I would also mention ```:py:class:`GeographyType` ``` here already and not only in the returns section?

jorisvandenbossche commented on src/accessors-geog.cpp:
```diff
@@ -63,70 +47,66 @@ double perimeter(PyObjectGeography a, double radius = numeric_constants::EARTH_R
 
 void init_accessors(py::module& m) {
+    py::options options;
+    options.disable_function_signatures();
```
It is not possible to set this globally for all modules that will be added?

benbovy replied:
Yes probably better to set this once in `spherely.cpp` (the `options` lifetime would go through all `init_xxx` calls). I initially thought this would be set only for some functions or classes but I ended-up setting it for all functions.

jorisvandenbossche commented on src/geography.cpp:
```diff
@@ -410,7 +496,12 @@ void init_geography(py::module &m) {
+        Returns
+        -------
+        prepared : Geography or array
+            The same input Geography object(s) with an attached index.
```
When adding this, it might be worth explicitly mentioning in the docstring higher up that the function also updates the geographies in place, and not only the ones that are returned

jorisvandenbossche commented on src/predicates.cpp:
```diff
@@ -65,27 +65,37 @@ class TouchesPredicate {
+        Returns
+        -------
+        bool or array
+            Results of the ``intersects`` test.
```
Sidenote, and this is totally fine since you already added it! But personally I don't find it needed to strictly always have an _explanation_ of the result. It's good to know it returns a bool or boolean array, but that it is the result of `intersects` is IMO just duplicating what is already said above.

(thought the same about the `intersection` et al docstrings)

jorisvandenbossche commented on src/projections.cpp:
```diff
@@ -7,12 +7,55 @@
+        Lightweight wrapper for selecting common reference systems used to
+        project Geography points or vertices.
+
+        Cannot be instanciated directly.
```
```suggestion
        Cannot be instantiated directly.
```

jorisvandenbossche commented on src/geography.cpp:
```diff
@@ -262,17 +283,23 @@ void init_geography(py::module &m) {
 
     auto pygeography_types = py::enum_<GeographyType>(m, "GeographyType", py::arithmetic(), R"pbdoc(
-        The enumeration of Geography types
+        The enumeration of Geography types.
```
Not entirely sure why this happens, but for some reason this first line is "quoted" in the docstring, as if sphinx thinks it is indented (https://spherely--89.org.readthedocs.build/en/89/_api_generated/spherely.GeographyType.html)

jorisvandenbossche commented on src/accessors-geog.cpp:
```diff
@@ -63,70 +47,66 @@ double perimeter(PyObjectGeography a, double radius = numeric_constants::EARTH_R
           py::vectorize(&centroid),
-          py::arg("a"),
-          R"pbdoc(
+          py::arg("geography"),
```
While we are at it (renaming those single args), should we make those first arguments positional-only? (so we avoid anyone ever doing `centroid(geography=...)` (renaming this keyword name is really only meant for being descriptive in the docs, I would say?)

jorisvandenbossche reviewed (commented):
Great work! 

I added some comments while going through the individual commits (thanks for the nice separate commits and messages, that certainly made it easier to review!), but you can see what you address here.

benbovy commented:
@jorisvandenbossche I addressed your review comments in the last two commits. Do you want to review it before merging this? Ortherwise this is ready I think.

jorisvandenbossche reviewed (approved):
Looks good, thanks for the updates!
